### ISSUE TITLE
Going from 7.5.2 to 7.5.3 requires Android compileSdkVersion >= 34

### ISSUE THREAD

**Issue by wsamoht** [RevenueCat/purchases-capacitor#211]
I just updated this plugin to 7.5.3 in my app from 7.4.0. The gradle build is failing with this message:

```
[capacitor]         1.  Dependency 'androidx.core:core-ktx:1.12.0' requires libraries and applications that
[capacitor]         depend on it to compile against version 34 or later of the
[capacitor]         Android APIs.
```

I tracked it down to 7.5.3 of this plugin which updated `purchases-hybrid-common` to 9.8.0 which in turn updated `purchases-android` to 7.6.0. This updated `androidxCore` from [1.8.0 to 1.12.0](https://github.com/RevenueCat/purchases-android/compare/7.5.2...7.6.0#diff-697f70cdd88ba88fe77eebda60c7e143f6ad1286bca75017421e93ad84fb87df).

Shouldn't this be considered a breaking change?

Capacitor 5 is the latest released version that has `compileSdkVersion` set to `33`. Capacitor 6 still in Beta updates it to `34`.

**Comment by vegaro** [comment]
Hi @wsamoht, you're correct and we are sorry about it, this was an unintentional change. It should have definitely been a minor (same way the dependency got updated) or a major, but definitely not a patch, which is easy to get updated unintentionally.

You're correct and the change comes from that increase of androidxCore, which we had to increase in order to fix an issue in Android 14 in our Android SDK.

We're having internal conversations around steps we can take to ensure  we don't ship patches like this again in the future. We will be updating the changelog in that version until we figure out what we do. 

We were by the way able to reproduce the issue you're facing and were able to get the app to work by increasing these two versions in our tester app https://github.com/RevenueCat/purchases-capacitor/blob/main/example/purchase-tester/android/variables.gradle#L4. If you're unable to do that due to incompatibilities in your code, can you please stick to 7.5.2 while we figure this out?

Our apologies for the inconvenience.

**Comment by vegaro** [comment]
You should also test your app runs fine in Android 14 devices when targetting compileSdkVersion 34. There could be some changes that break at runtime. It's very typical there are behavioural changes in UI, or around permissions. In the case of Android 14 for example, there were some changes around broadcast receivers.

We've tested our plugin against Android 14 and so far what we know is that it works fine, but it's possible another plugin you use has not been adapted to run on Android 14. To give you an example, it looks like the share plugin has issues when targetting Android 34 (14) (and running in Android 14 devices) https://github.com/ionic-team/capacitor-plugins/issues/1823#issuecomment-1770670263

So in summary, it's also a good idea to verify your app works fine when updating the `targetSdkVersion`. Fwiw, Google will start requiring targeting Android 14 around August, and won't accept new versions of apps that don't target that version number.

**Comment by wsamoht** [comment]
Thanks for that info! And good catch on the share plugin. To play it safe, we'll just wait for Capacitor 6 to be released before updating any SDK versions. So we'll stay on `7.5.2` of this plugin until then.

Feel free to close this issue unless you want to keep it open until you figure out what you are going to do with versioning :)

**Comment by tonidero** [comment]
Note here that we reverted the related change in version 7.5.5 and have added some specific tests to catch this in the future. Closing this, and thanks for reporting!
